.github/workflows/test.yml

```diff
@@ -7,8 +7,16 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: "3.8"
+          python-version: "3.9"
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.HOME }}/.cache/pypoetry
+          key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
       - name: Install library
-        run: pip install -e .[dev]
-      - name: Run tests
-        run: pytest
+        run: |
+          pip install poetry
```

Comment:
i dont know if this can cause issue but i remember there was another default way to install